app/controllers/purchases_controller.rb

```diff
@@ -22,9 +26,16 @@ def create
     end
   end
 
+  def limit
+    @@purchase_limit = params[:commit].to_i
+    redirect_to purchases_path
+  end
+
   private
 
   def purchase_params
-    params.require(:purchase).permit(:date, :description, :amount, :payee, :subcategory)
+    if params[:purchase]
```

Comment:
What happens if this conditional is not included? Does the other functionality break things?

Reply:
Oh, good catch - that's a legacy of the approach I was taking before I wrote the limit route, and was having to pass both the limit and new purchases through the same action.